docs/hub/how-to-upstream.md

```diff
@@ -0,0 +1,263 @@
+The `Repository` class allows you to push models or other repositories to the Hub. `Repository` is a wrapper over Git and Git-LFS methods, so make sure you have Git-LFS installed (see [here](https://git-lfs.github.com/) for more instructions) and set up before you begin. If you are already familiar with common Git commands, then the `Repository` class should feel familiar. 
+
+### Clone a repository
+
+The `clone_from` parameter clones a repository from a Hugging Face model ID:
+
+```python
+>>> repo = Repository(local_dir="w2v2", clone_from="facebook/wav2vec2-large-960h-lv60")
+```
+
+This parameter can also clone a repository from a specified directory using a URL (if you are working offline, this parameter should be `None`):
+
+```python
+>>> repo = Repository(local_dir="huggingface-hub", clone_from="https://github.com/huggingface/huggingface_hub")
+```
+
+### Using a local clone
+
+Instantiate a `Repository` object with a path to a local Git clone or repository:
+
+```python
+>>> from huggingface_hub import Repository
+>>> repo = Repository(local_dir="<path>/<to>/<folder>")
+```
+
+### Commit and push to cloned repository
+
+If you want to commit or push to a cloned repository that belongs to you or your organizations:
+
+1. Log in to your Hugging Face account with the following command:
+
+   ```bash
+   huggingface-cli login
+   ```
+
+2. Instantiate your `Repository` class with the parameter `use_auth_token=True`:
+   
+   ```python
+   >>> repo = Repository(local_dir="my-model", clone_from="<user>/<model_id>", use_auth_token=True)
+   ```
+
+You can also attribute a Git username and email to a cloned repository by specifiying the `git_user` and `git_email` parameters. When users commit to that repository, Git will be aware of the commit author.
+
+```python
+>>> repo = Repository(
+...   "my-dataset", 
+...   clone_from="<user>/<dataset_id>", 
+...   use_auth_token=True, 
+...   repo_type="dataset",
+...   git_user="MyName",
+...   git_email="me@cool.mail"
+... )
+```
+
+### Branch
+
+Switch between branches with `git_checkout`. For example, if you want to switch from `branch1` to `branch2`:
+
+```python
+>>> repo = Repository(local_dir="huggingface-hub", clone_from="<user>/<dataset_id>", revision='branch1')
+>>> repo.git_checkout("branch2")
+```
+
+### Add
+
+Add a file to commit with `git_add`:
+
+```python
+>>> repo = Repository(local_dir="huggingface-hub", clone_from="<user>/<dataset_id>")
+>>> repo.git_add("my-awesome-file")
+```
+
+This method has an `auto-lfs-track` parameter that can be set to `True` if you want to automatically track large files (>10MB):
+
+```python
+>>> repo = Repository(local_dir="huggingface-hub", clone_from="<user>/<dataset_id>")
+>>> repo.git_add("my-awesome-file", auto_lfs_track=True)
+```
+
+### Commit
+
+Commit a file with `git_commit`:
+
+```python
+>>> repo = Repository(local_dir="huggingface-hub", clone_from="<user>/<dataset_id>")
+>>> repo.git_add(commit_message="Commit my-awesome-file to the Hub")
```

Comment:
This should be `git_commit` - I don't know if putting the repository instantiation above is the best, as this wouldn't be a real-world scenario: the user would have to clone (first line), then save their files, `git_add` and finally `git_commit` (second line).

I would either remove the first line, or show a full example

Reply:
Removed the first line to better align with real-world scenario